pandas/core/arrays/integer.py

```diff
@@ -700,11 +700,6 @@ def _maybe_mask_result(self, result, mask, other, op_name):
         op_name : str
         """
 
-        # may need to fill infs
-        # and mask wraparound
-        if is_float_dtype(result):
-            mask |= (result == np.inf) | (result == -np.inf)
```

Comment:
i think this line was at the root of #27829, so this might fix that

Reply:
I think you're correct.